website/static/js/components/socialshare.js

```diff
@@ -0,0 +1,52 @@
+/**
+* Mithril components for social sharing.
+*/
+'use strict';
+var $ = require('jquery');
+var m = require('mithril');
+
+var utils = require('js/components/utils');
+var required = utils.required;
+
+var ShareButtons = {
+    openLinkInPopup: function(href) {
+        window.open(href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,width=600,height=400');
+        return false;
+    },
+    view: function(ctrl, options) {
+        var file_name = encodeURIComponent(required(options, 'file_name'));
```

Comment:
Also, `file_name` is a misnomer; you are passing the node title to it. How about just `name`?